plugin/src/main/groovy/com/novoda/gradle/command/AdbTask.groovy

```diff
@@ -40,11 +39,12 @@ public class AdbTask extends org.gradle.api.DefaultTask {
     }
 
     protected void assertDeviceConnected() {
-        def id = getDeviceId()
-        Device device = pluginEx.devices().find { device -> device.id == id }
```

Comment:
Here we wanted to remove accessing the method `devices` in the extension. 
We had 2 options
- Either we had to move this `devices()` to somewhere else that is accessible. 
- Or we use `adb get-state` to check if the device is connected. 

We had all the info we need to use `get-state`, so we decided to do that.